crate/cli/src/actions/cover_crypt/decrypt.rs

```diff
@@ -38,9 +47,18 @@ impl DecryptAction {
         f.read_to_end(&mut data)
             .with_context(|| "Fail to read the file to decrypt")?;
 
+        // Recover the unique identifier or set of tags
+        let id = if let Some(key_id) = &self.user_key_id {
+            key_id.clone()
+        } else if let Some(tags) = &self.tags {
+            serde_json::to_string(&tags)?
+        } else {
+            cli_bail!("Either --key-id or one or more --tag must be specified")
+        };
```

Comment:
Because 
1- this is Clap CLI code and enums are not trivial with Clap. Check the locate command if you want an idea of what it takes.
2- enum does not solve the issue at hand here: testing that we have one clap option --key-id or multiple specifications of --tag option

Reply:
I get the 1. For 2, it seems to me that using something like solves it:
```
enum KeyDesignator {
     Id(Id),
    Tags(Vec<Tag>),
}
```
You could then write this in an implementation of `to_string()` for `KeyDesignator`:
```
match self.designator {
     Id(id) => id.clone(),
     Tags(tags) => serde_json::to_string(&tags)?,
}
```